Add multisearch configuration

kulturbande opened the pull request:
- added default multisearch configuration to search also for partial words
- added documentation for the Gem configuration

tvdeyen commented on README.md:
````diff
@@ -41,6 +41,23 @@ Simply pass `searchable: false` to your content definitions and Alchemy will not
+Configure the gem in an initializer. The default configurations are:
+
+```ruby
+   Alchemy::PgSearch.config = {
````
```suggestion
   ::PgSearch.config = {
```

tvdeyen commented on lib/alchemy/pg_search/config.rb:
```diff
@@ -2,7 +2,12 @@ module Alchemy
       @@config = {
-        paginate_per: 10
+        paginate_per: 10,
+        search_options: { # https://github.com/Casecommons/pg_search#searching-using-different-search-features
```
```suggestion
```

tvdeyen commented on lib/alchemy/pg_search/engine.rb:
```diff
@@ -16,6 +16,9 @@ class Engine < ::Rails::Engine
         Alchemy.publish_targets << Alchemy::PgSearch::IndexPageJob
+
+        # configure multiselect to find also partial words
+        ::PgSearch.multisearch_options = Alchemy::PgSearch.config[:search_options]
```
```suggestion
        ::PgSearch.multisearch_options ||= {using: {
             tsearch: { prefix: true }
           }}
```

tvdeyen reviewed (commented):
We should not proxy the `pg_search` config through `Alchemy`

tvdeyen commented on README.md:
````diff
@@ -41,6 +41,29 @@ Simply pass `searchable: false` to your content definitions and Alchemy will not
+         
+Configure the gem in an initializer. The default configurations are:
+
+```ruby
+   Alchemy::PgSearch.config = {
+      paginate_per: 10, # amount of results per page
+  }
+```
````
The indentation is a bit off here

```suggestion
```ruby
Alchemy::PgSearch.config = {
  paginate_per: 10, # amount of results per page
}
```
```

tvdeyen reviewed (approved):
Love that. A tiny nit, but then we can merge this

kulturbande committed:
Add multisearch configuration

- added default multisearch configuration to search also for partial words
- added documentation for the Gem configuration
